### PR TITLE
feat(android, ios): continuous update for ListView scrolling event

### PR DESCRIPTION
**Pull request by m1ga**
**JIRA:** https://jira.appcelerator.org/browse/AC-6722

When `continuousUpdate` is enabled the `scrolling` event will fire for every visibleItem/Section change. 

**Test:**
* start example app
* start scrolling
* in the top right corner (or log) you'll see the current index/section
* click "change" button to enable/disable `continuousUpdate`

disabled or with 10.1.1GA you only see an update when you change scroll direction.

**Example**
```js
var toggle = true;
var sections = [];
const win = Ti.UI.createWindow();
const btn = Ti.UI.createButton({
	title: "change",
	bottom: 20
});
const lbl = Ti.UI.createLabel({
	top: 10,
	right: 10,
	width: Ti.UI.SIZE,
	height: Ti.UI.SIZE
})
const listView = Ti.UI.createListView({
	height: Ti.UI.FILL,
	width: Ti.UI.FILL,
	continuousUpdate: true
});

btn.addEventListener("click", function() {
	toggle = !toggle;
	listView.continuousUpdate = toggle
	console.log(listView.continuousUpdate);
})

for (var s = 0; s < 5; s++) {
	var section = Ti.UI.createListSection({
		headerTitle: 'Section ' + s
	});
	var set = [];
	for (var i = 0; i < 20; ++i) {
		set.push({
			properties: {
				title: 'Item 0 ' + i
			}
		})
	}
	section.setItems(set);
	sections.push(section);
}

listView.sections = sections;
win.add([listView, btn, lbl]);

listView.addEventListener("scrolling", function(e) {
	lbl.text = "Item: " + e.firstVisibleItemIndex + " section:" + e.firstVisibleSectionIndex;
	console.log("Item", e.firstVisibleItemIndex, e.firstVisibleSectionIndex);
})
win.open();
```

### CODE REVIEW

**Comment by jquick-axway** [comment]
@m1ga , did you run clang-format on the Java code?

You don't need to do that anymore. We use Java "checkstyles" which will trigger a build error for anything that violates our coding guidelines and explain what the issue is. Also, Android Studio's build log will show a clickable link that will take you directly to the line of code that violated the rule.

**Comment by m1ga** [comment]
Thanks @jquick-axway. I saw those warnings e.g. line too long and then I thought I'll run clang to automatically fix those warnings. Didn't know that it's not needed anymore. Was a quick automatic way of fixing all issues. 
I've fixed the unneeded changes

File: apidoc/Titanium/UI/ListView.yml
Comment (jquick-axway) [inline apidoc/Titanium/UI/ListView.yml:908]: Would you mind documenting what the use-case for enabling this is in the description please?
I'm curious why you want this feature. And I'm willing to help add it to iOS.

Also, the following sentence is awkward.
`...event will everytime a new item is visible.`
I think the following is better. _(Not trying to be picky.)_
`...event will fire every time a new item is scrolled in.`

Comment (m1ga) [reply apidoc/Titanium/UI/ListView.yml:908]: Sure, will update the docs tomorrow. I need to know where I'm at to change a menu e.g. which element just came into view. Currently I'm using markers but they only fire once so I have to add them again when I've hit one. 
Especially iOS had problems when I've add the markers and it filled the list after adding them. It did fire the marker event for all markers already because it was triggered before the actual data was visible and my 2nd (normally offscreen marker) was visible.
Having the scoll event just telling me the whole time which element is visible (fistVisibleItem/section) I can easily use that for lazy loading, dynamically switching menus (indicator where you are) and other triggers (e.g. changing a headline)

**Comment by m1ga** [comment]
@hansemannn could you have a look at this one.

File: build/.idea/.gitignore
Comment (hansemannn) [inline build/.idea/.gitignore:3]: This file should be removed

Comment (m1ga) [reply build/.idea/.gitignore]: sorry, removed it.

**Comment by hansemannn** [comment]
The code looks fine - all additions to the code are guarded / in separate parts. Same for iOS. Have you tested iOS properly as well? And for custom cells?

**Comment by hansemannn** [comment]
In any case, I would prioritize bug fixes first at this stage and do features / improvements after that, with a prio on features that have full parity.

**Comment by m1ga** [comment]
tested in on iOS too with the example and an app with a ListView with 4 templates, 5 sections. And since its an older PR the whole iOS/Android test-suite was done too.

The new parts are hidden behind a `false` by default property so it won't affect any existing apps. Only if you set `continuousUpdate:true` it will make any difference. But it can wait for other PRs :smile:

**Comment by hansemannn** [comment]
I feel like it's not the most frequently asked-for feature, so let's approve it already, but merge after the fixes have all been merged.

**Review by hansemannn — APPROVED** [review]
iOS looks good!

**Comment by hansemannn** [comment]
@m1ga Happy to take this before 12, e.g. 11.1.0!

**Comment by m1ga** [comment]
> @m1ga Happy to take this before 12, e.g. 11.1.0!

ah nice! Wasn't sure :smile: It's covered by a test case too so from a functional state point quick to test/merge

**Review by hansemannn — APPROVED** [review]

**Pull request merged**
